### PR TITLE
refactor(core): Make fwdRecord default in Handler

### PR DESCRIPTION
**Pull request by timoffex**
Description
---
Makes `fwdRecord` the default case in Handler and documents all exceptions.

I double checked against `wandb_internal.proto` that every single record type is covered by the switch.

### CODE REVIEW

File: core/pkg/server/handler.go [hunk 1]
```diff
@@ -167,43 +167,27 @@ func (h *Handler) Do(inChan <-chan *spb.Record) {
...
-		case *spb.Record_Preempting:
-			h.fwdRecord(record)
-		case *spb.Record_Stats:
-			h.fwdRecord(record)
-		case *spb.Record_Summary:
-			h.fwdRecord(record)
-		case *spb.Record_Telemetry:
-			h.fwdRecord(record)
-		case *spb.Record_UseArtifact:
+		default:
```
Comment (kptkin) [inline core/pkg/server/handler.go:170]: lets move to the bottom?

**Review by kptkin — APPROVED** [review]

File: core/pkg/server/handler.go (hunk 1 above)
Comment (timoffex) [reply core/pkg/server/handler.go:170]: I would say it's better to have the exceptions at the bottom and the normal case at the top. It's different in spirit from using `default` as a fallback, which is why it's normally at the bottom.

File: core/pkg/server/handler.go [hunk 2]
```diff
@@ -167,43 +167,27 @@ func (h *Handler) Do(inChan <-chan *spb.Record) {
...
 		case *spb.Record_Final:
+			// Deprecated.
 		case *spb.Record_Footer:
+			// Deprecated.
 		case *spb.Record_Header:
+			// The record's VersionInfo gets modified before forwarding.
 		case *spb.Record_NoopLinkArtifact:
+			// Deprecated.
 		case *spb.Record_Tbrecord:
+			// Never forwarded.
```
Comment (kptkin) [inline core/pkg/server/handler.go:186]: wait why?

File: core/pkg/server/handler.go (hunk 1 above)
Comment (kptkin) [reply core/pkg/server/handler.go:170]: yes, but i think it is more golang idiomatic so i would still think it's better at the bottom could add a comment if you want to clarify that this is the default behavior

File: core/pkg/server/handler.go (hunk 2 above)
Comment (kptkin) [reply core/pkg/server/handler.go:186]: i see that in the python code we do forward it

File: core/pkg/server/handler.go (hunk 1 above)
Comment (timoffex) [reply core/pkg/server/handler.go:170]: i think the keyword `default` already implies it's the default behavior :) and anyway this is a common pattern:

* https://mtsknn.fi/blog/switch-statements-default-doesnt-have-to-be-the-last-case/#normal-case-first
* https://docs.datadoghq.com/code_analysis/static_analysis_rules/go-best-practices/switch-default-first-or-last/ (see the second "good" example)

File: core/pkg/server/handler.go (hunk 2 above)
Comment (kptkin) [reply core/pkg/server/handler.go:186]: tho, it looks like we don't do anything in the sender with the record 🤷

Comment (timoffex) [reply core/pkg/server/handler.go:186]: yeah, it's just not used. I'll double check that it's OK to not forward it and then update in a different PR (I have more coming for Request records)

File: core/pkg/server/handler.go (hunk 1 above)
Comment (kptkin) [reply core/pkg/server/handler.go:170]: if that makes you happy

File: core/pkg/server/handler.go (hunk 2 above)
Comment (timoffex) [reply core/pkg/server/handler.go:186]: ooh, glad I checked actually, it's not safe to forward unless we add the corresponding case in the Sender

Comment (kptkin) [reply core/pkg/server/handler.go:186]: yep, i saw it is just noop in the sender in the python code

Comment (kptkin) [reply core/pkg/server/handler.go:186]: it's kinda weird in that case, seems that it should be a request and not a record

Comment (timoffex) [reply core/pkg/server/handler.go:186]: i agree! surfacing these issues is the point of the refactor :) now it's easy to see what the next cleanup steps are

**Pull request merged**